samza-kv/src/main/scala/org/apache/samza/storage/kv/BaseKeyValueStorageEngineFactory.scala

```diff
@@ -124,22 +135,56 @@ trait BaseKeyValueStorageEngineFactory[K, V] extends StorageEngineFactory[K, V]
       new LoggedStore(rawStore, changeLogSystemStreamPartition, collector, loggedStoreMetrics)
     }
 
-    // wrap with serialization
-    val serializedMetrics = new SerializedKeyValueStoreMetrics(storeName, registry)
-    val serialized = new SerializedKeyValueStore[K, V](maybeLoggedStore, keySerde, msgSerde, serializedMetrics)
+    var toBeAccessLoggedStore: KeyValueStore[K, V] = null
+
+    if (largeMessagesExpected) {
```

Comment:
The `getStorageEngine` method is already pretty long, and it's a little hard to keep track of all of the different wrapped stores. Please consider pulling out some of your new logic to other helper methods for readability. Also, it might allow you to share some methods with the "dropLargeMessage" case.

Reply:
Done. Thanks!